src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -1135,7 +1135,7 @@ class Peer(
                                 else -> null
                             }
                         }
-                        offerManager.receiveMessage(msg, remoteChannelUpdates, currentTipFlow.filterNotNull().first().first)?.let {
+                        offerManager.receiveMessage(msg, remoteChannelUpdates, currentTipFlow.filterNotNull().first())?.let {
```

Comment:
This one is a bit annoying, because one of the motivations for this PR is to make phoenix wake-up lighter in the context of responding to a BOLT12 invoice request by removing the dependency on the blockchain. Unfortunately we need  the current blockheight here:

https://github.com/ACINQ/lightning-kmp/blob/8b6ac2e906e11835a1bc0fa7978463b372e6af64/src/commonMain/kotlin/fr/acinq/lightning/payment/OfferManager.kt#L159

Is there any workaround here @thomash-acinq?

Reply:
Yes, we could remove the whole `RouteBlindingEncryptedDataTlv.PaymentConstraints` TLV. It would deviate from the spec but it's fine because only the trampoline node would see it (we would need to modify the trampoline node to accept a missing `PaymentConstraints` in this case).

Reply:
I don't think we would gain much by removing the dependency on the blockchain here, because we will need it anyway when receiving the HTLC as highlighted by [your other comment](https://github.com/ACINQ/lightning-kmp/pull/650/commits/8b6ac2e906e11835a1bc0fa7978463b372e6af64#r1620712981).

Since we need a block height when receiving the HTLC, which should come quickly after the `invoice_request`, the only thing we would save by not needing a block height here would potentially be a small latency boost is we're slower to get the block height than we are to connect to ACINQ and receive the onion message, but that's the only thing we'd gain. And it isn't latency that is visible to the user, since they're receiving a payment (potentially in the background), not sending one. So do we really care?

Reply:
Maybe I should just drop the blockheight commit altogether?

Reply:
A good first step seems indeed to be to only take the first two commits for now, while we work on a different solution to obtain the block height.